UM/CentralFileStorage.py

```diff
@@ -5,101 +5,126 @@
-    directory at every version upgrade.
+    This class stores and retrieves items (files or directories) stored in a location central to all versions of the
+    application. Its purpose is to provide a way for plug-ins to store big items (files or directories) without having
+    those big items copied to the new resource directory at every version upgrade.
 
-    Each file gets a file ID and a version number by which to identify the individual file. The file ID and version
-    number combination needs to be unique for a file and together refer to a specific file's contents. When retrieving
-    the file, the retriever needs to specify the hash of the file that it expects to find, and this hash is checked
-    before giving up the location of the file. This hash will guard against unauthorised changes to the file. The
-    plug-in, if properly signed, will then not be surprised by malicious content in that file.
+    Each items gets a path ID and a version number by which to identify the individual items. The path ID and version
+    number combination needs to be unique for an item and together refer to a specific item's contents. When retrieving
+    the item, the retriever needs to specify the hash of the item that it expects to find, and this hash is checked
+    before giving up the location of the item. This hash will guard against unauthorised changes to the item. The
+    plug-in, if properly signed, will then not be surprised by malicious content in that file or directory.
 
-    If multiple files need to be moved, it's also possible to add a file called "central_storage.json" to the root
+    If multiple items need to be moved, it's also possible to add a file called "central_storage.json" to the root
     directory of the plugin. This json needs to contain a list, of which each element is a list of 4 elements.
-    The first element holds the relative path to the file, the second is the ID, the third is the ID of the file and the
-    last item is the hash.
+    The first element holds the relative path to the item (file or directory), the second is the ID, the third is the
+    ID of the item and the last item is the hash.
 
     A brief example of such a file:
     [
-        ["files/VeryLargeFileToStore.big", "very_large_file", "1.0.1", "abcdefghijklmnop"]
+        ["files/VeryLargeFileToStore.big", "very_large_file", "1.0.1", "abcdefghijklmnop"],
+        ["relative/path/to/dir", "relative/path/to/dir/in/storage", "1.0.1, "124986978cb21e"]
     ]
     """
 
+    # In some cases a plugin might ask for files to be moved, but it's not needed (since the plugin is actually bundled
+    # In order to ensure that the same API can still be used, we store those situations.
+    _unmoved_files = {}  # type: Dict[str, str]
+
     @classmethod
-    def store(cls, file_path: str, file_id: str, version: Version = Version("1.0.0")) -> None:
```

Comment:
You renamed `file_id` to `path_id`, but I think you misunderstood that parameter. The file ID is only a name of the item. It's not necessarily the same as the file name or file path (since files might be renamed in later versions).

A better name for it would be `item_id`, since that is how you are calling it in the documentation now.